src/steamship/data/plugin/index_plugin_instance.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Dict, Optional
+
+from pydantic import Field
+
+from steamship import SteamshipError
+from steamship.base.client import Client
+from steamship.base.model import CamelModel
+from steamship.data.embeddings import EmbeddingIndex
+from steamship.data.plugin.plugin_instance import PluginInstance
+
+
+class EmbedderInvocation(CamelModel):
+    """The parameters capable of creating/fetching an Embedder (Tagger) Plugin Instance."""
+
+    plugin_handle: str
+    instance_handle: Optional[str] = None
+    config: Optional[Dict[str, Any]] = None
+    version: Optional[str] = None
+    fetch_if_exists: bool = True
+
+
+# Hard coded list of plugin handles that the Steamship client will use to create
+# instance of this object instead of a normal PluginInstance.
+SHIMMED_INDEX_PLUGIN_HANDLES = ["embedding-index"]
+
+
+class EmbeddingIndexPluginInstance(PluginInstance):
```

Comment:
Would it be a good idea to re-define the methods of `EmbeddingIndex` in this class? 

For example we would add 

```
def search(
        self,
        query: Union[str, List[str]],
        k: int = 1,
        include_metadata: bool = False,
    ) 
    self.index.search(query, k, include_metadata)
```

Benefit: I won't have to call `index.index` to operate with the index, instead I'm going to use the pluginInstance directly, similar to how we define .blockify on the level of the Blockifier

Reply:
Ignore what I said above. After re-reading the description, I realize you'll address this in another PR

Reply:
👍 Your comment is right-on though.

And yeah, I'm trying to do this across several PRs in an attempt to make each change easy to see from you & Dave's perspective since this is a pretty big shift in surface area. Next pr begins to do exactly what you suggest